packages/core/src/events/Events.tsx

```diff
@@ -52,17 +52,17 @@ export const Events: React.FC = ({ children }) => {
 
   // Disable/Enable handlers when the enabled state is toggled
   useEffect(() => {
-    if (!handlersRef.current) {
+    if (!handlers) {
       return;
     }
 
     if (!enabled) {
-      handlersRef.current.disable();
+      handlers.disable();
       return;
     }
 
-    handlersRef.current.enable();
-  }, [enabled]);
+    handlers.enable();
+  }, [enabled, handlers]);
```

Comment:
I think it would be enough to add `handlersRef.current` to the dependency array:

```tsx
  // Disable/Enable handlers when the enabled state is toggled
  useEffect(() => {
    if (!handlersRef.current) {
      return;
    }

    if (!enabled) {
      handlersRef.current.disable();
      return;
    }

    handlersRef.current.enable();
  }, [enabled, handlersRef.current]);
```

Reply:
My tests seem to confirm, that it is enough to add `handlersRef.current` to the dependency array

Reply:
Is it good practice though? I don't think we should depend on ref values as dependencies? https://github.com/facebook/react/issues/16121

Reply:
Wouldn't that be the case for `useMemo` then?

```tsx
  const memoizedStore = React.useMemo(() => store, [store]);
  const memoizedHandlers = React.useMemo(() => handlers, [handlers]);

  // Generate new handlers whenever the editor deserializes a new state
  useEffect(() => {
    // TODO: Remove this when we refactor the EditorState
    actions.history
      .ignore()
      .setState((state) => (state.handlers = handlersFactory(memoizedStore)));

    return () => {
      if (!memoizedHandlers) {
        return;
      }

      memoizedHandlers.cleanup();
    };
  }, [actions, handlersFactory, hydrationTimestamp]);

  // Disable/Enable handlers when the enabled state is toggled
  useEffect(() => {
    if (!memoizedHandlers) {
      return;
    }

    if (!enabled) {
      memoizedHandlers.disable();
      return;
    }

    memoizedHandlers.enable();
  }, [enabled, memoizedHandlers]);
```

But... why do we need to use `ref` or `memo` anyways? 🤔 

Since we do
```tsx
  const storeRef = useRef(store);
  storeRef.current = store;
  const handlersRef = useRef(handlers);
  handlersRef.current = handlers;
```
and I cannot find a place where we would need to have access to "reffed" version of the store or handlers? Am I missing something? 🤔

Reply:
Yeah I need the ref values here: https://github.com/prevwong/craft.js/blob/e250f449841feb007c4e54448dbdc8b57d39bba0/packages/core/src/events/Events.tsx#L41-L49

If I use `memoizedStore` and `memoizedHandlers` like in your example - then I will need to add those as a dependency to that `useEffect` hook, but that will cause that hook to execute everytime `handlers` and `store` is updated.

Reply:
Thanks. LGTM then 🙂 👍